pkg/storage/storage.go

```diff
@@ -193,8 +193,6 @@ func (s *Storage) LockRelease(name string) error {
 // If release doesn't exist or wasn't previously locked - the unlock will pass
 func (s *Storage) UnlockRelease(name string) {
 	s.Log("unlocking release %s", name)
-	s.releaseLocksLock.Lock()
```

Comment:
Oh, wait... there is a race condition if we totally remove this, but leave the other lock. It is possible that the `releaseLocks` map will get concurrent writes. I'll mark where that can happen.